src/link/MachO/Dylib.zig

```diff
@@ -71,9 +72,53 @@ pub fn closeFile(self: Dylib) void {
     }
 }
 
+pub fn decodeArch(cputype : macho.cpu_type_t) !std.Target.Cpu.Arch {
+    const arch: std.Target.Cpu.Arch = switch (cputype) {
+        macho.CPU_TYPE_ARM64 => .aarch64,
+        macho.CPU_TYPE_X86_64 => .x86_64,
+        else => |value| {
+            log.err("unsupported cpu architecture 0x{x}", .{value});
+            return error.UnsupportedCpuArchitecture;
+        },
+    };
+    return arch;
+}
+
 pub fn parse(self: *Dylib) !void {
     log.debug("parsing shared library '{s}'", .{self.name.?});
 
+    var library_offset : u64 = 0;
+    {
+        const fat_header = try self.file.?.reader().readStructEndian(macho.fat_header, builtin.Endian.Big);
+        if(fat_header.magic == macho.FAT_MAGIC) {
```

Comment:
If you want to be extra-correct you could detect all the valid `FAT_*` numbers here and report an error if the endianness/format doesn't match the only format you're parsing here.

Reply:
The fat headers are _always_ big endian. I don't know if it's worth checking for the incorrect endianess as that would imply something has gone horribly wrong with the source file?

Reply:
My bad, I missed that part in the `fat.h` header:
```
* All structures defined here are always written and read to/from disk
* in big-endian order.
```
Carry on then, and sorry for the noise!

Reply:
I've added a comment to the code though now to make this clearer :)